content/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-the-codeql-workflow-for-compiled-languages.md

```diff
@@ -135,3 +135,34 @@ For more information about the `if` conditional, see "[Workflow syntax for GitHu
 For more tips and tricks about why `autobuild` won't build your code, see "[Troubleshooting the {% data variables.product.prodname_codeql %} workflow](/code-security/secure-coding/troubleshooting-the-codeql-workflow)."
 
 If you added manual build steps for compiled languages and {% data variables.product.prodname_code_scanning %} is still not working on your repository, contact {% data variables.contact.contact_support %}.
+
+## Excluding directories from the database by customizing the build steps
+
+Sometimes you do not want to build and analyze your whole project and its dependencies. You can omit build dependencies from the {% data variables.product.prodname_codeql %} database by separating their build from the main build that's tracked by {% data variables.product.prodname_codeql %}.
+
+### Example
+
+The following example illustrates this approach for a C++ project that's built using `make`, but the procedure applies to all compiled languages and build systems.
+
+Your project tree contains three subdirectories, `src/`, `lib1`, and `lib2` and your default build command, `make all`, builds them all. If you configure the workflow with the default build command, it generates a {% data variables.product.prodname_codeql %} database with the code and dependencies of all three subdirectories.
+
+```sh
+codeql database create cpp-database --language=cpp --command='make all'
+```
+
+You can omit one of the subdirectories and its dependencies from the {% data variables.product.prodname_codeql %} database by building it before calling the {% data variables.product.prodname_codeql %} CLI.
+
+```sh
+make lib2 
+
+codeql database create cpp-database --language=cpp --command='make src lib1'
+```
```

Comment:
Would this be an appropriate replacement for someone using the CodeQL action instead of the CLI?

````suggestion
```yml
    steps:
    - name: Checkout repository
      uses: actions/checkout@v2
      
    - name: Initialize CodeQL
      uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}

    - run: |
       make all

    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```

You can omit a subdirectory and its dependencies from the {% data variables.product.prodname_codeql %} database by building it before the "Initialize {% data variables.product.prodname_codeql %}" step.

```yml
    steps:
    - name: Checkout repository
      uses: actions/checkout@v2
      
    - name: Build lib2
     - run: |
       make lib2

    - name: Initialize CodeQL
      uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}

    - run: |
       make src lib1

    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```
````

Reply:
The action makes this more brittle.  In the first configuration, remove the 
```
    - run: |
       make all
```
part, that will be run by codeql.

In the second configuration, remove 
```
    - run: |
       make src lib1
```
because that will also be run by the codeql runner.

Reply:
Thank you. Would I need to put the autobuild part back instead?

That is:
```
    - name: Initialize CodeQL
	    uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}
	
    - name: Autobuild
      uses: github/codeql-action/autobuild@v1
	
    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```

Reply:
Thanks.

````suggestion
```yml
    steps:
    - name: Checkout repository
      uses: actions/checkout@v2
      
    - name: Initialize CodeQL
      uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}

    # Autobuild attempts to build any compiled languages
    # In this case it runs the default build command `make all`
    - name: Autobuild
      uses: github/codeql-action/autobuild@v1

    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```

You can omit a subdirectory and its dependencies from the {% data variables.product.prodname_codeql %} database by building it before the "Initialize {% data variables.product.prodname_codeql %}" step.

```yml
    steps:
    - name: Checkout repository
      uses: actions/checkout@v2
      
    - name: Build lib2
     - run: |
       make lib2

    - name: Initialize CodeQL
      uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}

    # Autobuild attempts to build any compiled languages
    # In this case it skips making `lib2` as this is already built
    - name: Autobuild
      uses: github/codeql-action/autobuild@v1

    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```
````